address-space-controller/src/main/java/io/enmasse/controller/auth/AuthController.java

```diff
@@ -132,6 +132,11 @@ public String getDefaultCertProvider() {
 
     @Override
     public AddressSpace reconcile(AddressSpace addressSpace) throws Exception {
+
+        if ( Controller.isDeleted(addressSpace)) {
+            return addressSpace;
```

Comment:
Adding this to every controller feels like something that can be easily missed. Maybe another option would be to add this to the loop in ControllerChain.java ? This would ensure it cannot be missed if we're adding another one.

Consider instead doing the finalizing in reconcileAll in the finalizer controller, which is called at the end, where I think its ok to skip this check.

Reply:
I though so as well, I would rather add this to the controller interface, and let implementors decide if they want to handle this.

Reply:
Hmm, in what do you mean? I was thinking of modifiying ControllerChain.java like this:

```
for address space in list:
  if (!Controller.isDeleted(addressSpace)) {
    for (controller in chain) {
      contorller.reconcile(address space)
    }
  }
```

Reply:
I pushed a change of what I had in mind.

Reply:
@ctron LGTM!